src/ASM/LR/ASMu2D.C

```diff
@@ -475,103 +479,77 @@ bool ASMu2D::generateFEMTopology ()
 }
 
 
-bool ASMu2D::connectPatch (int edge, ASM2D& neighbor, int nedge, bool revers,
-                           int basis, bool coordCheck, int thick)
+bool ASMu2D::connectPatch (int edge, ASM2D& neighbor, int nedge,
+                           bool revers, int, bool coordCheck, int thick)
 {
-  return false;
+  ASMu2D* un = dynamic_cast<ASMu2D*>(&neighbor);
+  if (!un)
+    return false;
+
+  if (!this->connectBasis(edge,*un,nedge,revers,1,0,0,coordCheck,thick))
+    return false;
+
+  this->addNeighbor(un);
+  return true;
 }
 
 
-/*
 bool ASMu2D::connectBasis (int edge, ASMu2D& neighbor, int nedge, bool revers,
-                           int basis, int slave, int master)
+                           int basis, int slave, int master,
+                           bool coordCheck, int thick)
 {
-  // Set up the slave node numbers for this surface patch
-
-  int n1, n2;
-  if (!this->getSize(n1,n2,basis)) return false;
-  int node = slave+1, i1 = 0;
-
-  switch (edge)
-    {
-    case 2: // Positive I-direction
-      node += n1-1;
-    case 1: // Negative I-direction
-      i1 = n1;
-      n1 = n2;
-      break;
-
-    case 4: // Positive J-direction
-      node += n1*(n2-1);
-    case 3: // Negative J-direction
-      i1 = 1;
-      break;
-
-    default:
-      std::cerr <<" *** ASMu2D::connectPatch: Invalid slave edge "
-                << edge << std::endl;
-      return false;
-    }
+  if (this->isShared() && neighbor.isShared())
+    return true;
+  else if (this->isShared() || neighbor.isShared())
+  {
+    std::cerr <<" *** ASMu2D::connectBasis: Logic error, cannot"
+	      <<" connect a shared patch with an unshared one"<< std::endl;
+    return false;
+  }
 
-  int i;
-  IntVec slaveNodes(n1,0);
-  for (i = 0; i < n1; i++, node += i1)
-    slaveNodes[i] = node;
+  // Set up the slave node numbers for this surface patch
+  IntVec slaveNodes;
+  this->getBoundaryNodes(edge, slaveNodes, basis, thick, true);
+  for (int& it : slaveNodes)
+    it += slave;
 
   // Set up the master node numbers for the neighboring surface patch
+  IntVec masterNodes;
+  neighbor.getBoundaryNodes(nedge, masterNodes, basis, thick, true);
```

Comment:
Edge nodes returned from LRSplineSurface::getEdgeFunctions( ) are not sorted, so will in general differ on master surface and slave surface of the edge. Either sort them manually inside the getBoundaryNodes-call or LRSplineSurface::edgeCurve( ) which is slightly more costly since it creates a SplineCurve interpolation object, but it does return the basis functions in sorted order.